python/stempy/image/__init__.py

```diff
@@ -26,68 +33,74 @@ def create_stem_images(input, inner_radii, outer_radii, scan_dimensions=(0, 0),
+
+            imgs = _image.create_stem_images(reader.begin(), reader.end(),
+                                             inner_radii, outer_radii,
+                                             scan_dimensions, center)
     else:
-        reader = input
-
-    imgs = _image.create_stem_images(reader.begin(), reader.end(),
-                                     inner_radii, outer_radii,
-                                     scan_dimensions, center)
+        raise Exception('Type of input, ' + str(type(input)) +
+                        ', is not known to stempy.image.create_stem_images()')
 
     images = [np.array(img, copy=False) for img in imgs]
     return np.array(images, copy=False)
 
-def create_stem_histogram(numBins, reader, inner_radii,
-                          outer_radii, scan_dimensions=(0, 0),
-                          center=(-1, -1)):
-    """Create a histogram of the stem images generated from the input.
-
-    :param numBins: the number of bins the histogram should have.
-    :param reader: the file reader that has already opened the data.
-    :type reader: stempy.io.reader
-    :param inner_radii: a list of inner radii. Must match
-                        the length of `outer_radii`.
-    :type inner_radii: list of ints
-    :param outer_radii: a list of outer radii. Must match
-                        the length of `inner_radii`.
-    :type outer_radii: list of ints
-    :param scan_dimensions: the dimensions of the scan, where the order is
-                            (width, height). If set to (0, 0), an attempt
-                            will be made to read the scan dimensions from
-                            the data file.
-    :type scan_dimensions: tuple of ints of length 2
-    :param center: the center of the images, where the order is (x, y). If set
-                   to (-1, -1), the center will be set to
-                   (scan_dimensions[0] / 2, scan_dimensions[1] / 2).
-    :type center: tuple of ints of length 2
-
-    :return: The bins and the frequencies of the histogram.
-    :rtype: a tuple of length 2 of lists
-    """
-    # create stem images
-    imgs = _image.create_stem_images(reader.begin(), reader.end(),
-                                     inner_radii, outer_radii,
-                                     scan_dimensions, center)
-
-    allBins = []
-    allFreqs = []
-    for inImage in imgs:
-        bins = _image.get_container(inImage, numBins)
-        freq = _image.create_stem_histogram(inImage, numBins, bins)
-        bins = np.array(bins, copy=False)
-        freq = np.array(freq, copy=False)
-        allBins.append(bins)
-        allFreqs.append(freq)
-
-    return allBins, allFreqs
-
-# This one exists for backward compatibility
+@deprecation.deprecated(
+    deprecated_in='1.0', removed_in='1.1',
+    details='Use stempy.image.create_stem_images() instead.')
 def create_stem_image(reader, inner_radius, outer_radius,
```

Comment:
I do wonder is we should just keep this one around ( forward the call on ). Thoughts?